src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/SqlColumnEncryptionAzureKeyVaultProvider.cs

```diff
@@ -310,6 +348,49 @@ private byte[] CompileMasterKeyMetadata(string masterKeyPath, bool allowEnclaveC
             return Encoding.Unicode.GetBytes(masterkeyMetadata.ToLowerInvariant());
         }
 
+        /// <summary>
+        /// Converts the numeric value of each element of a specified array of bytes to its equivalent hexadecimal string representation.
+        /// </summary>
+        /// <param name="source">An array of bytes to convert.</param>
+        /// <returns>A string of hexadecimal characters</returns>
+        /// <remarks>
+        /// Produces a string of hexadecimal character pairs preceded with "0x", where each pair represents the corresponding element in value; for example, "0x7F2C4A00".
+        /// </remarks>
+        private string ToHexString(byte[] source)
+        {
+            if (source == null)
+            {
+                return null;
+            }
+
+            return "0x" + BitConverter.ToString(source).Replace("-", "");
+        }
+
+        /// <summary>
+        /// Returns the cached decrypted data encryption key, or unwraps the encrypted data encryption if not present.
+        /// </summary>
+        /// <param name="encryptedDataEncryptionKey">Encrypted Data Encryption Key</param>
+        /// <param name="createItem">The delegate function that will decrypt the encrypted column encryption key.</param>
+        /// <returns>The decrypted data encryption key.</returns>
+        /// <remarks>
+        ///
+        /// </remarks>
+        private byte[] GetOrCreateColumnEncryptionKey(string encryptedDataEncryptionKey, Func<byte[]> createItem)
+        {
+            return columnEncryptionKeyCache.GetOrCreate(encryptedDataEncryptionKey, createItem);
```

Comment:
Is encryptedDataEncryptionKey sufficient to uniquely identify the key in the cache? In the global cache in the driver, we've included the database server name along with some other attributes. When this updated AKV provider is used as a global provider, how will this cache behave?